ucrop/src/main/java/com/yalantis/ucrop/task/BitmapCropTask.java

```diff
@@ -149,19 +151,36 @@ private boolean crop() throws IOException {
         Log.i(TAG, "Should crop: " + shouldCrop);
 
         if (shouldCrop) {
-            ExifInterface originalExif = new ExifInterface(mImageInputPath);
             saveImage(Bitmap.createBitmap(mViewBitmap, cropOffsetX, cropOffsetY, mCroppedImageWidth, mCroppedImageHeight));
             if (mCompressFormat.equals(Bitmap.CompressFormat.JPEG)) {
-                ImageHeaderParser.copyExif(originalExif, mCroppedImageWidth, mCroppedImageHeight, mImageOutputPath);
+                if (mImageInputUri != null && "content".equals(mImageInputUri.getScheme()) && "content".equals(mImageOutputUri.getScheme())) {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                        ImageHeaderParser.copyExif(mContext.get(), mCroppedImageWidth, mCroppedImageHeight, mImageInputUri, mImageOutputUri);
+                    } else {
+                        Log.e(TAG, "It is not possible to write exif info into file represented by \"content\" Uri if Android < LOLLIPOP");
+                    }
+                } else if (mImageInputUri != null && "content".equals(mImageInputUri.getScheme())) {
+                    ImageHeaderParser.copyExif(mContext.get(), mCroppedImageWidth, mCroppedImageHeight, mImageInputUri, mImageOutputPath);
```

Comment:
It seems that build version check is missed here

Reply:
No it is not. The previous build version check is because `mImageOutputUri` was been used instead of `mImageOutputPath`. When I use an URI as source to build an `ExifInterface` I have to open a *seekable* filedescriptor and this is not possible before Lollipop. I mention it in the proper `copyExif` overloads documentations.
In this particular case, we use a path to build an `ExifInterface` and than we have no such limitation.